src/main/java/io/github/deansg/jeocodio/GeocodioClient.java

```diff
@@ -132,28 +157,57 @@ private String formatFieldsParam(List<String> fields) {
     }
 
     private <T> CompletableFuture<T> sendAsync(HttpRequest httpRequest, Class<T> clazz) {
-        return httpClient.sendAsync(httpRequest, HttpResponse.BodyHandlers.ofString())
-                .thenApply(resp -> {
-                    var body = resp.body();
-                    if (resp.statusCode() != 200) {
-                        throw new GeocodioStatusCodeException(resp.statusCode(), body);
-                    }
-                    return body;
-                })
+        return httpClient.sendAsync(httpRequest, HttpResponse.BodyHandlers.ofInputStream())
+                .thenApply(this::readResponse)
                 .thenApply(str -> this.gson.fromJson(str, clazz));
     }
 
+    private String readResponse(HttpResponse<InputStream> resp) {
+        try (var inputStream = getResponseInputStream(resp)) {
+            String json = new String(inputStream.readAllBytes(), StandardCharsets.UTF_8);
+            if (resp.statusCode() != 200) {
+                throw new GeocodioStatusCodeException(resp.statusCode(), json);
+            }
+            return json;
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```

Comment:
According to Sonarqube and other places, it is bad practice to throw a generic exception.

https://rules.sonarsource.com/java/RSPEC-112/

Maybe a new exception should be created? `GeocodioReadException` which derives from `RuntimeException`.

Reply:
Replaced with UncheckedIOException, I think it's the most appropriate for this case